.github/actions/deploy-environment/action.yml

```diff
@@ -62,7 +62,23 @@ runs:
     - id: terraform
       run: |
         make ci ${{ inputs.environment_name }} terraform-apply
+         cd terraform
+        OUTPUTS=($(terraform output --json | jq -r 'keys | @sh' | tr -d \'))
+        for o in "${OUTPUTS[@]}"
+        do
+          echo "${o}=$(terraform output -raw ${o})" >> $GITHUB_ENV
```

Comment:
Is this required?

Reply:
Yes, it is required.
The reason the scripts produce the output from terraform:
 output "functionapp_name" {
  value = azurerm_linux_function_app.function_app.name
}
and this value is used here: app-name: ${{ env.functionapp_name }}
 uses: Azure/functions-action@v1
      with:
        app-name: ${{ env.functionapp_name }}
        package: ${{ github.workspace }}/artifact

TO USE FUNCTION APP NAME FOR DEV, TEST, PRE-PROD AND PRODUCTION ENVIRONMENTS.